From 58eae92049fb5acfd4f134fb7e33197c6c986af0 Mon Sep 17 00:00:00 2001
From: Jeff King <peff@peff.net>
Date: Mon, 7 Feb 2011 17:05:35 -0500
Subject: [PATCH] check both system and user cache for emptiness

We will use whichever is the more recent of the two caches.
However, the early check for "is the cache empty" could
uselessly trigger if the system cache was empty but the user
cache was not.

Signed-off-by: Jeff King <peff@peff.net>
---
 apt-file |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/apt-file b/apt-file
index ad4f6bf..a9f8d46 100755
--- a/apt-file
+++ b/apt-file
@@ -774,7 +774,8 @@ sub main {
     my $sources = parse_sources_list $Conf->{sources_list};
     errorx "No valid sources in @{$Conf->{sources_list}}" if !defined $sources;
 
-    if ( $Conf->{is_search} && dir_is_empty( $Conf->{cache} ) ) {
+    if ( $Conf->{is_search} && dir_is_empty( $Conf->{cache} )
+         && defined $Conf->{user_cache} && dir_is_empty( $Conf->{user_cache} ) ) {
         errorx "The cache directory is empty. You need to run 'apt-file update' first.";
     }
 
-- 
1.7.4.rc2.27.gd0787

